### PR TITLE
fix: null pointer error project data source

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR fixes a null pointer error: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5807776023/job/15743326685#step:4:52

Jira Ticket: [INTMDB-1002](https://jira.mongodb.org/browse/INTMDB-1002)

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/data_source_mongodbatlas_projects.go [hunk 1]
```diff
@@ -242,23 +242,28 @@ func flattenProjects(ctx context.Context, client *MongoDBClient, projects []*mat
...
+				"cluster_count":             project.ClusterCount,
+				"created":                   project.Created,
+				"region_usage_restrictions": project.RegionUsageRestrictions,
+				"teams":                     flattenTeams(teams),
+				"api_keys":                  flattenAPIKeys(apiKeys),
+				"limits":                    flattenLimits(limits),
 			}
+
+			if projectSettings != nil {
+				resultEntry["is_collect_database_specifics_statistics_enabled"] = projectSettings.IsCollectDatabaseSpecificsStatisticsEnabled
```
Comment (blva) [inline mongodbatlas/data_source_mongodbatlas_projects.go:258]: [q] just double-checking as I don't have full context, is it OK to not fill this data? do we need to fill empty fields by any chance?

**Review by blva — APPROVED** [review]

File: mongodbatlas/data_source_mongodbatlas_projects.go (hunk 1 above)
Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_projects.go:258]: no, we don't have to. This is what is stored in the terraform state file which does not require to fill all the fields of the resource

**Comment by maastha** [comment]
For my knowledge, what can cause the projectSettings to be null?

**Comment by andreaangiolillo** [comment]
> For my knowledge, what can cause the projectSettings to be null?

errors from the API or NOT_FOUND. The resource logic is defined in such a way that we do not return an error in the READ if one of the API calls fails

**Review by Zuhairahmed — APPROVED** [review]

**Pull request merged**